docs/developer-guide/custom-widgets/preact-widgets.md

```diff
@@ -0,0 +1,189 @@
+
+```json
+{
+  "compilerOptions": {
+    "jsx": "react-jsx",
+    "jsxImportSource": "preact"
+  }
+}
+```
+
+> Note: Developer environments vary. Refer to the [Preact Typescript](https://preactjs.com/guide/v10/typescript) documentation for additional environments.
+
+## Example: Layer List Widget with Preact
+
+Below is a comprehensive example demonstrating a layer list widget implemented using Preact for dynamic UI rendering:
+
+```tsx
+import {
+  _deepEqual as deepEqual,
+  _applyStyles as applyStyles,
+  _removeStyles as removeStyles
+} from '@deck.gl/core'
+import type {
+  Deck, Viewport, Widget, WidgetPlacement, Layer
+} from '@deck.gl/core'
+import {render} from 'preact';
+
+type LayerListWidgetProps = {
+  id?: string;
+  /**
+   * Widget positioning within the view. Default: 'top-left'.
+   */
+  placement?: WidgetPlacement;
+  /**
+   * View to attach to and interact with. Required when using multiple views. Default: null
+   */
+  viewId?: string | null;
+  /**
+   * CSS inline style overrides.
+   */
+  style?: Partial<CSSStyleDeclaration>;
+  /**
+   * Additional CSS class.
+   */
+  className?: string;
+};
+
+class LayerListWidget implements Widget<LayerListWidgetProps> {
+  id = 'layer-list-widget';
+  props: LayerListWidgetProps;
+  placement: WidgetPlacement = 'top-left';
+  viewId?: string | null = null;
+  viewports: {[id: string]: Viewport} = {};
+  layers: Layer[] = [];
+  deck?: Deck<any>;
+  element?: HTMLDivElement;
+
+  constructor(props: LayerListWidgetProps) {
+    this.id = props.id ?? this.id;
+    this.placement = props.placement ?? this.placement;
+    this.viewId = props.viewId ?? this.viewId;
+
+    this.props = {
+      ...props,
+      style: props.style ?? {}
+    };
+  }
+
+  onAdd({deck}: {deck: Deck<any>}): HTMLDivElement {
+    const {style, className} = this.props;
+    const element = document.createElement('div');
+    element.classList.add('deck-widget', 'deck-widget-layer-list');
+    if (className) element.classList.add(className);
+    applyStyles(element, style);
+    this.deck = deck;
+    this.element = element;
+    this.update();
+    return element;
+  }
+
+  setProps(props: Partial<LayerListWidgetProps>) {
+    const oldProps = this.props;
+    const el = this.element;
+    // Handle when CSS changes.
+    if (el) {
+      if (oldProps.className !== props.className) {
+        if (oldProps.className) el.classList.remove(oldProps.className);
+        if (props.className) el.classList.add(props.className);
+      }
+      if (!deepEqual(oldProps.style, props.style, 1)) {
+        removeStyles(el, oldProps.style);
+        applyStyles(el, props.style);
+      }
+    }
+    // Handle when props change.
+    this.placement = props.placement ?? this.placement;
+    this.viewId = props.viewId ?? this.viewId;
+    Object.assign(this.props, props);
+    this.update();
+  }
+
+  onRedraw({layers}: {layers: Layer[]}) {
+    this.layers = layers;
+    this.update();
+  }
```

Comment:
@Pessimistress one thing I noticed is this includes all sub-layers in a flattened list. While that could be useful sometime, I imagine most widgets would want to display state from the top-level layers. Is there a way to access this aside from `deck.props.layers`?